### PR TITLE
[Feature] Added support for metadata sidecar for quota limit feature

### PR DESCRIPTION
**Pull request by shefali-malhotra**
# Description
Added csi-metadata-retriever sidecar support in powerscale 2.7.0 for quota  limit feature. 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/742 |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [X] Installed latest powersacle driver using csm operator
csm_operator_pods

- verify driver events:
csm_operator_driver_events

### CODE REVIEW

**Review by HarishH-DELL — APPROVED** [review]

File: samples/storage_csm_powerscale_v270.yaml [hunk 1]
```diff
@@ -287,7 +287,14 @@ spec:
           # Default value: powerstore
           - name: "X_CSI_REPLICATION_CONTEXT_PREFIX"
             value: "powerscale"
-      
+
+      - name: metadataretriever
+        # image: Image to use for csi-metadata-retriever. This shouldn't be changed
+        image: dellemc/csi-metadata-retriever:v1.3.0
```
Comment (rensyct) [inline samples/storage_csm_powerscale_v270.yaml:293]: if the image should not be changed, please remove it from the samples

File: samples/storage_csm_powerscale_v270.yaml [hunk 2]
```diff
@@ -287,7 +287,14 @@ spec:
...
+        envs:
+          - name: "CSI_RETRIEVER_ENDPOINT"
+            value: "/var/run/csi/csi_retriever.sock"
```
Comment (rensyct) [inline samples/storage_csm_powerscale_v270.yaml:296]: Can this value be changed? if yes, is that tested. I dont see the corresponding code for that

**Review by rensyct — CHANGES_REQUESTED** [review]

File: operatorconfig/driverconfig/common/default.yaml [hunk 3]
```diff
@@ -24,4 +24,7 @@ images:
   externalhealthmonitorcontroller: registry.k8s.io/sig-storage/csi-external-health-monitor-controller:v0.8.0
   
   # "images.sdcmonitor" defines the container images used to monitor sdc container
-  sdcmonitor: dellemc/sdc:3.6.0.6
\ No newline at end of file
+  sdcmonitor: dellemc/sdc:3.6.0.6
+
+  #"images.metadataretriever" defines the container images used for csi metadata retriever
```
Comment (rensyct) [inline operatorconfig/driverconfig/common/default.yaml:29]: Please add this sidecar in all K8s samples supported by 2.7.0 CSI driver

File: samples/storage_csm_powerscale_v270.yaml (hunk 2 above)
Comment (shefali-malhotra) [reply samples/storage_csm_powerscale_v270.yaml:296]: retriever endpoint value is fixed

File: operatorconfig/driverconfig/common/default.yaml (hunk 3 above)
Comment (shefali-malhotra) [reply operatorconfig/driverconfig/common/default.yaml:29]: updated the files .

**Review by nitesh3108 — APPROVED** [review]

**Review by rensyct — APPROVED** [review]

**Comment by rensyct** [comment]
@shefali-malhotra , please rebase your branch

**Pull request merged**